### PR TITLE
Test delete custom fields

### PR DESCRIPTION
**Pull request by Bharani0012**
added unit test file for delete custom field api

updated delete custom field api endpoint

### CODE REVIEW

File: .github/workflows/integrationtests.yml [hunk 1]
```diff
@@ -100,7 +100,7 @@ jobs:
       - name: Run Python Integration Tests
         working-directory: ${{ github.workspace }}
         run: |
-          python -m unittest discover -s ./test/IntegrationTests
+          python -m unittest discover -s ./test/IntegrationTests -p "*.py"
```
Comment (SaravanakumarR2018) [inline .github/workflows/integrationtests.yml:103]: Try making this change as 

python -m unittest discover -s ./test/IntegrationTests -pattern "test_*.py"

**Review by SaravanakumarR2018 — COMMENTED** [review]
Lets make sure we print the testcase function names in setup and teardown for all test files like below


def setUp(self):
        # Log the start of a test
        logger.info(f"Starting test: {self._testMethodName}")
    
    def tearDown(self):
        # Log the end of a test
        logger.info(f"Finished test: {self._testMethodName}")
        
        
        looks like you have put the same changes as the earlier Pull request for list and create custom fields
        
        Merge that list and create custom fields Pull Request first and then create this pull request on top so that we dont have to review the same changes again

File: test/IntegrationTests/TicketService/test_delete_custom_field.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+)
+logger = logging.getLogger(__name__)
+
+class TestDeleteCustomFieldAPI(unittest.TestCase):
+    BASE_URL = "http://localhost:8000"
+
+    def setUp(self):
+        """Setup function to initialize valid customer GUID and custom field."""
+        logger.info("=== Initializing test setup ===")
+
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/TicketService/test_delete_custom_field.py:19]: add a function name in setup and tear down function

def setUp(self):
        # Log the start of a test
        logger.info(f"Starting test: {self._testMethodName}")
    
    def tearDown(self):
        # Log the end of a test
        logger.info(f"Finished test: {self._testMethodName}")

File: test/IntegrationTests/TicketService/test_delete_custom_field.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "field_name": "test_field",
+            "field_type": "VARCHAR(255)",
+            "required": True
+        }
+        response = requests.post(custom_field_url, json=data)
+        self.assertEqual(response.status_code, HTTPStatus.CREATED, "Failed to add custom field during setup")
+
+    def test_delete_custom_field_success(self):
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/TicketService/test_delete_custom_field.py:38]: Lets add all types of custom fields

INT BOOLEAN FLOAT VARCHAR(255) etc and then verify them by listing all of them and checking and then lets test deleting them as a testcase one by one, and then list all the custom fields one by one and verify the deleted custom field is not present

Add 2 or 3 custom fields and delete one of them and verify the remaining 2 custom fields are present

Try giving a wrong URL  like

{self.BASE_URL}/custom_fields/{'test_field'}/blah/blah/blah?customer_guid={self.valid_customer_guid}" and test that this URL should not work. It should not work by default, as FastAPI takes care of all these routing but you can test once as a testcase

**Review by SaravanakumarR2018 — COMMENTED** [review]
You have the changes of create and list custom fields also on this PR

First merge the create and list custom fields PR and then create a PR for delete custom fields so that we can review only the delete custom fields PR

I have also provided comments for delete custom fields API

File: test/IntegrationTests/TicketService/test_delete_custom_field.py (hunk 3 above)
Comment (Bharani0012) [reply test/IntegrationTests/TicketService/test_delete_custom_field.py:38]: modified sir

File: .github/workflows/integrationtests.yml (hunk 1 above)
Comment (Bharani0012) [reply .github/workflows/integrationtests.yml:103]: updated sir

Comment (Bharani0012) [reply .github/workflows/integrationtests.yml:103]: i got error when i use like this sir 

Run python -m unittest discover -s ./test/IntegrationTests -pattern "test_*.py"

**Pull request closed without merging**